### DISCUSSION TITLE
Limited availability for the next couple of days

### DISCUSSION THREAD

**Discussion by markusjnagel** [sonnen-tools/node-red#19] [Announcements]
Hey folks,

first of all, a huge THANK YOU for your feedback, your ideas, sharing your results, etc... that helps a ton to make this better and iron out some bugs. I really appreciate this!!

![image](https://github.com/sonnen-tools/node-red/assets/33287465/e78069e7-f82c-4051-ad4b-10e7a060ed78)

So, I will implement a new, additional, hopefully better way of deciding what to charge and when - and when best to discharge. However, I'm pretty booked in my dayjob for the next couple of days, so please expect a delay in responses and improved features.... 

Just to let you know, I haven't lost interest - but the next couple of days (until and including next Wed, 17th Jan, 2024) will be tough and leave me no time for this. 

Thanks for your patience! :smile: 

Thanks,

Markus

**Comment by dlindh** [comment]
I hope you are ok Markus! If you lost interest in development I would still like to thank you for what has already been produced :) Helps me a lot :)

**Comment by doncuco** [comment]
Hey Markus,
I hope you are doing well! Just wanted to thank you for your work and hope you can soon rejoin us :)
In the meantime I am working on my on fork that recognizes the PV Input so that the price gets adjusted with cheap solar energy.
Best regards from south Germany and happy easter holidays!

**Reply by stefan227** [reply]
hey doncuco, have you got something to share? It seems Markus is of this project. i would like o see if there is some progress. Greetings from northern germany

**Reply by doncuco** [reply]
Not much, nur i decided to try to work on a few things. For example i Chance the fact that when battery Price is NaN the battery stays in Manuel Mode which Blocks from discharging. In that case i allow discharging.
And as i run nodered in Home Assistant i right now send some Sensors to Home Assistant to See the battery Energy Price 
Next Step would be to take care Ofen the solarinput.
And in my case wäre i have high sonsumption at the Night, the Script is nearly Never fully charging my battery. So i Need to fix that.

**Reply by Olli510** [reply]
> And in my case wäre i have high sonsumption at the Night, the Script is nearly Never fully charging my battery. So i Need to fix that.

I experienced a similar problem. My Sonnen-battery quits charging after an hour, even if there are more cheap hours following.

**Reply by stefan227** [reply]
doncuco, would you mind telling where to change that? how to tell the battery NaN is good for discharging? Is NaN the real value or is it some sort of placeholder for values below 0?
